src/main/java/models/task/TaskList.java

```diff
@@ -147,46 +147,117 @@ public int getSize() {
+                    successMessages.add(results.get(0));
+                    results.remove(0);
+                }
+                results.remove(0); //remove "error start"
+                errorMessages.addAll(results);
+            }
         }
         newTaskRequirementDetails.remove(0);
-        addTaskRequirements(taskIndexNumber, newTaskRequirementDetails);
+
+        if (newTaskRequirementDetails.size() != 0) {
+            successMessages.addAll(addTaskRequirements(taskIndexNumber, newTaskRequirementDetails));
+        }
+
+        //combining success messages and error messages into one array
+        if (successMessages.size() == 1) {
+            successMessages.remove(0);
+        } else {
+            successMessages.add("");
+        }
+        if (errorMessages.size() == 1) {
+            successMessages.add("There were no errors!");
+        } else {
+            successMessages.addAll(errorMessages);
+        }
+
+        return successMessages.toArray(new String[0]);
     }
 
     /**
      * Removes task requirement with input index.
      * @param taskIndexNumber Index of task to be edited.
      * @param taskReqIndexesToBeRemoved String containing indexes of task requirements to be removed.
+     * @return String array of success and error messages
      */
-    private void removeTaskRequirements(int taskIndexNumber, String taskReqIndexesToBeRemoved) {
+    private ArrayList<String> removeTaskRequirements(int taskIndexNumber, String taskReqIndexesToBeRemoved) {
         String[] indexesToBeRemoved = taskReqIndexesToBeRemoved.split(" ");
+        ArrayList<String> successMessages = new ArrayList<>();
+        ArrayList<String> errorMessagesForInvalidIndexes = new ArrayList<>();
         Arrays.sort(indexesToBeRemoved);
+        ArrayList<Integer> listOfUsedIntegers = new ArrayList<>();
+        int startingNumOfTaskRequirements = this.taskList.get(taskIndexNumber - 1).getNumOfTaskRequirements();
         for (int i = indexesToBeRemoved.length - 1; i >= 0; i--) {
-            int indexToBeRemoved = Integer.parseInt(indexesToBeRemoved[i]);
-            if (indexToBeRemoved > 0
-                    && indexToBeRemoved <= this.taskList.get(taskIndexNumber - 1).getNumOfTaskRequirements()) {
-                //input validation if input index to be removed does not exist, but currently does not show mistake to
-                // user
-                this.taskList.get(taskIndexNumber - 1).removeTaskRequirement(indexToBeRemoved);
+            try {
+                int indexToBeRemoved = Integer.parseInt(indexesToBeRemoved[i]);
+                if (indexToBeRemoved > 0
+                        && indexToBeRemoved <= startingNumOfTaskRequirements) {
+                    if (!listOfUsedIntegers.contains(indexToBeRemoved)) {
+                        successMessages.add("'"
+                                + this.taskList.get(taskIndexNumber - 1).getTaskRequirements().get(indexToBeRemoved)
```

Comment:
I think there might be an error here. TaskList.get(task index) already accounts for the -1 if I'm not wrong!
![image](https://user-images.githubusercontent.com/42560942/68085069-ac7dce00-fe77-11e9-938f-defd61d75d1c.png)

Reply:
oh this one i used the get() method from ArrayList class, not the getTask() function